engine/src/main/java/io/camunda/zeebe/engine/processing/resource/ResourceDeletionProcessor.java

```diff
@@ -178,4 +186,13 @@ private NoSuchResourceException(final long resourceKey) {
       super(String.format(ERROR_MESSAGE_RESOURCE_NOT_FOUND, resourceKey));
     }
   }
+
+  private static final class RunningInstancesException extends IllegalStateException {
+    private static final String ERROR_MESSAGE_RUNNING_INSTANCES =
+        "Expected to delete resource but there are still running instances '%s'";
+
+    private RunningInstancesException(final List<Long> runningInstanceKeys) {
+      super(String.format(ERROR_MESSAGE_RUNNING_INSTANCES, runningInstanceKeys.toString()));
+    }
+  }
```

Comment:
🔧 We might want to use `active instances` as the more common terminology that our users may be familiar with